src/modules/ekf2/EKF2.cpp

```diff
@@ -490,37 +493,37 @@ void EKF2::Run()
 			}
 		}
 
-		if (_optical_flow_sub.updated()) {
-			_new_optical_flow_data_received = true;
-			optical_flow_s optical_flow;
-
-			if (_optical_flow_sub.copy(&optical_flow)) {
-				flowSample flow {};
-				// NOTE: the EKF uses the reverse sign convention to the flow sensor. EKF assumes positive LOS rate
-				// is produced by a RH rotation of the image about the sensor axis.
-				flow.flow_xy_rad(0) = -optical_flow.pixel_flow_x_integral;
-				flow.flow_xy_rad(1) = -optical_flow.pixel_flow_y_integral;
-				flow.gyro_xyz(0) = -optical_flow.gyro_x_rate_integral;
-				flow.gyro_xyz(1) = -optical_flow.gyro_y_rate_integral;
-				flow.gyro_xyz(2) = -optical_flow.gyro_z_rate_integral;
-				flow.quality = optical_flow.quality;
-				flow.dt = 1e-6f * (float)optical_flow.integration_timespan;
-				flow.time_us = optical_flow.timestamp;
-
-				if (PX4_ISFINITE(optical_flow.pixel_flow_y_integral) &&
-				    PX4_ISFINITE(optical_flow.pixel_flow_x_integral) &&
-				    flow.dt < 1) {
-
-					_ekf.setOpticalFlowData(flow);
-				}
+		bool new_optical_flow_data_received = false;
```

Comment:
Good point, I tried that before but it didn't work because it wasn't in sync with `ekf_updated`. I saw that you moved the `new_optical_flow_data_received`  out of `if (ekf_updated)` so it should work properly now.